### PR TITLE
Use GCC instead of gompi for OpenBLAS in goolf-1.4.10 test toolchain.

### PR DESCRIPTION
**Pull request by bartoldeman**
This allows to use the golf toolchain within goolf-1.4.10, also added.

### CODE REVIEW

File: test/framework/easyconfigs/test_ecs/s/ScaLAPACK/ScaLAPACK-2.0.2-gompi-1.4.10-OpenBLAS-0.2.6-LAPACK-3.4.2.eb [hunk 1]
```diff
@@ -20,7 +20,7 @@ blassuff = '-LAPACK-3.4.2'
 
 versionsuffix = "-%s-%s%s" % (blaslib, blasver, blassuff)
 
-dependencies = [(blaslib, blasver, blassuff)]
+dependencies = [(blaslib, blasver, blassuff, ('GCC', '4.7.2'))]
```
Comment (boegel) [inline test/framework/easyconfigs/test_ecs/s/ScaLAPACK/ScaLAPACK-2.0.2-gompi-1.4.10-OpenBLAS-0.2.6-LAPACK-3.4.2.eb]: @bartoldeman There should be no need to hardcode to `GCC` here, see most recent `ScaLAPACK` easyconfigs like `ScaLAPACK-2.0.2-gompic-2018b-OpenBLAS-0.3.1.eb`.

Comment (bartoldeman) [reply test/framework/easyconfigs/test_ecs/s/ScaLAPACK/ScaLAPACK-2.0.2-gompi-1.4.10-OpenBLAS-0.2.6-LAPACK-3.4.2.eb]: did my other comments get lost?

Comment (boegel) [reply test/framework/easyconfigs/test_ecs/s/ScaLAPACK/ScaLAPACK-2.0.2-gompi-1.4.10-OpenBLAS-0.2.6-LAPACK-3.4.2.eb]: What other comments? Maybe that answers your question. ;)

**Comment by boegel** [comment]
@bartoldeman Hmm, not sure why the tests fail with `Irresolvable dependencies encountered: OpenBLAS/0.2.6-gompi-1.3.12-LAPACK-3.4.2 (at easybuild/tools/robot.py:351 in resolve_dependencies)` now... I didn't expect that to happen (clearly).

**Comment by bartoldeman** [comment]
Essentially it boils down to this:
with the explicit GCC,4.7.2:
```
[oldeman@build-node test_ecs]$ eb --include-easyblocks=../../sandbox/easybuild/easyblocks/t/toy.py --try-toolchain=gompi,1.3.12 --robot=$PWD --force --dry-run ScaLAPACK-2.0.2-gompi-1.4.10-OpenBLAS-0.2.6-LAPACK-3.4.2.eb
== temporary log file in case of crash /tmp/eb-3CpT2G/easybuild-Ja2ZQ0.log
Dry run: printing build status of easyconfigs and dependencies
 * [ ] /home/oldeman/easybuild-framework/test/framework/easyconfigs/test_ecs/g/GCC/GCC-4.7.2.eb (module: Core | gcc/4.7.2)
 * [ ] /home/oldeman/easybuild-framework/test/framework/easyconfigs/test_ecs/o/OpenBLAS/OpenBLAS-0.2.6-GCC-4.7.2-LAPACK-3.4.2.eb (module: avx2/Compiler/gcc4.7 | openblas/0.2.6)
 * [ ] /home/oldeman/easybuild-framework/test/framework/easyconfigs/test_ecs/g/GCC/GCC-4.6.4.eb (module: Core | gcc/4.6.4)
 * [ ] /home/oldeman/easybuild-framework/test/framework/easyconfigs/test_ecs/h/hwloc/hwloc-1.6.2-GCC-4.6.4.eb (module: avx2/Compiler/gcc4.6 | hwloc/1.6.2)
 * [ ] /home/oldeman/easybuild-framework/test/framework/easyconfigs/test_ecs/o/OpenMPI/OpenMPI-1.6.4-GCC-4.6.4.eb (module: avx2/Compiler/gcc4.6 | openmpi/1.6.4)
 * [ ] /home/oldeman/easybuild-framework/test/framework/easyconfigs/test_ecs/g/gompi/gompi-1.3.12.eb (module: Core | gompi/1.3.12)
 * [ ] /tmp/eb-3CpT2G/tweaked_easyconfigs/ScaLAPACK-2.0.2-gompi-1.3.12-OpenBLAS-0.2.6-LAPACK-3.4.2.eb (module: avx2/MPI/gcc4.6/openmpi1.6 | scalapack/2.0.2)
== Temporary log file(s) /tmp/eb-3CpT2G/easybuild-Ja2ZQ0.log* have been removed.
== Temporary directory /tmp/eb-3CpT2G has been removed.
```
but without it:
```
[oldeman@build-node test_ecs]$ eb --include-easyblocks=../../sandbox/easybuild/easyblocks/t/toy.py --try-toolchain=gompi,1.3.12 --robot=$PWD --force --dry-run ScaLAPACK-2.0.2-gompi-1.4.10-OpenBLAS-0.2.6-LAPACK-3.4.2.eb
== temporary log file in case of crash /tmp/eb-55PY71/easybuild-JkYU8f.log
ERROR: Failed to determine minimal toolchain for dep OpenBLAS 0.2.6-LAPACK-3.4.2
```

**Comment by bartoldeman** [comment]
I have pushed a revert.

File: test/framework/easyconfigs/test_ecs/f/FFTW/FFTW-3.3.3-GCC-4.7.2-serial.eb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+sources = [SOURCELOWER_TAR_GZ]
+source_urls = [homepage]
+
+common_configopts = "--enable-openmp --with-pic"
+
+configopts = [
+    common_configopts + " --enable-single --enable-sse2 --enable-mpi",
+    common_configopts + " --enable-long-double --enable-mpi",
+    common_configopts + " --enable-quad-precision",
+    common_configopts + " --enable-sse2 --enable-mpi",  # default as last
```
Comment (akesandgren) [inline test/framework/easyconfigs/test_ecs/f/FFTW/FFTW-3.3.3-GCC-4.7.2-serial.eb]: A bit of nitpicking... --enable-mpi should not be used in the serial FFTW build.
I know this is a test config that should never be built, but someone might find it and use it as an example...

Comment (bartoldeman) [reply test/framework/easyconfigs/test_ecs/f/FFTW/FFTW-3.3.3-GCC-4.7.2-serial.eb]: ok. done

**Comment by akesandgren** [comment]
Thanks @bartoldeman

**Pull request merged**